pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,820 @@
+        Claimed(T::AccountId, Balance),
+        /// The rewards are recorded
+        RewardsRecorded(T::AccountId, Balance),
+        /// The slash is recorded
+        SlashRecorded(T::AccountId, Balance),
+
+        DepositEventToRelaychain(T::AccountId, EraIndex, StakingOperationType, Balance),
+        /// Bond operation in relaychain was successed.
+        BondSucceed(EraIndex),
+        /// Unbond operation in relaychain was successed.
+        UnbondSucceed(EraIndex),
+    }
+
+    /// The exchange rate converts staking native token to voucher.
+    #[pallet::storage]
+    #[pallet::getter(fn exchange_rate)]
+    pub type ExchangeRate<T: Config> = StorageValue<_, Rate, ValueQuery>;
+
+    /// Fraction of staking currency currently set aside for insurance pool
+    #[pallet::storage]
+    #[pallet::getter(fn reserve_factor)]
+    pub type ReserveFactor<T: Config> = StorageValue<_, Ratio, ValueQuery>;
+
+    /// The total amount of insurance pool.
+    #[pallet::storage]
+    #[pallet::getter(fn insurance_pool)]
+    pub type InsurancePool<T: Config> = StorageValue<_, Balance, ValueQuery>;
+
+    /// The total amount of staking pool.
+    #[pallet::storage]
+    #[pallet::getter(fn staking_pool)]
+    pub type StakingPool<T: Config> = StorageValue<_, Balance, ValueQuery>;
+
+    /// Store total stake amount and total unstake amount during current era,
+    /// And will update when trigger new era, calculate whether bond/unbond/rebond,
+    /// Will be remove when corresponding era has been finished success.
+    #[pallet::storage]
+    #[pallet::getter(fn matching_pool)]
+    pub type MatchingPool<T: Config> =
+        StorageMap<_, Blake2_128Concat, EraIndex, MatchingPoolBuffer, ValueQuery>;
+
+    /// Store single user's stake and unstake request during each era,
+    ///
+    /// For stake request, after all xtoken have been mint, this may take 6 hours in Kusama, and one day in Polkadot
+    /// The data can be removed after user claim their xToken
+    /// the corresponding era's operation in `StakingOperationHistory` should also be successed.
+    ///
+    /// For unstake request, xtoken need to be burn, and get token back,
+    /// the data can be removed after the lock period passed and user claim.
+    /// this may take 7 days in Kusama, and 28 days in Polkadot.
+    /// the corresponding era's operation in `StakingOperationHistory` should also be successed.
+    #[pallet::storage]
+    #[pallet::getter(fn matching_queue)]
+    pub type MatchingQueue<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        T::AccountId,
+        Blake2_128Concat,
+        EraIndex,
+        MatchingUserBuffer,
+        ValueQuery,
+    >;
+
+    /// Previous era index on Parachain.
+    ///
+    /// Considering that in case stake client cannot update era index one by one.
+    #[pallet::storage]
+    #[pallet::getter(fn previous_era)]
+    pub type PreviousEra<T: Config> = StorageValue<_, EraIndex, ValueQuery>;
```

Comment:
ValueQuery will give by default era 0, let's check if relaychain starts from `0` or `1`

Reply:
@MrPai

Reply:
in relaychain, the default value is 0;
https://github.com/paritytech/substrate/blob/master/frame/staking/src/pallet/mod.rs#L712